### ISSUE TITLE
ToolStripButton should not support both Invoke and Toggle patterns

### ISSUE THREAD

**Issue by dmitrii-drobotov** [dotnet/winforms#9108]
### .NET version

.NET 8

### Did it work in .NET Framework?

Yes

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

Yes, the regression was introduced in #8961 and affects only .NET 8 

### Issue description

Accessibility Insights report following error for checked/checkable ToolStripButton:
![A screenshot of test result in Accessibility Insights reporting error on ToolStripButton and showing that it should not support both Invoke and Toggle patterns](https://github.com/dotnet/winforms/assets/102954094/c9ff5ed1-8e24-4eda-aed4-05740a32b9e0)

ToolStripMenuItem also supports both Invoke and Toggle patterns, but AI doesn't have this error because it applies only to "button" control type.

### Steps to reproduce

1. Run WinformsControlTest app from this repo and open "ToolStrips" form, or run sample app: [ToolStripButtonsTest.zip](https://github.com/dotnet/winforms/files/11190660/ToolStripButtonsTest.zip).
2. Open Accessibility Insights and test the form with it

**Comment by Olina-Zhang** [comment]
Verified this issue in the latest .Net 8.0 SDK build: 8.0.100-preview.5.23274.2, it is fixed, see following test results in AccessibilityInsight and Inspect:
**AccessibilityInsight:**
![image](https://github.com/dotnet/winforms/assets/26474449/176d57e2-718d-43a3-a830-dd7692e53e82)
**Inspect:**
![image](https://github.com/dotnet/winforms/assets/26474449/a368a17c-c315-4d3a-97a9-78d66ccc104f)

**Comment by Olina-Zhang** [comment]
@dmitrii-drobotov From my testing result in Inspect as following, when toolStripButton's CheckOnClick or Checked property is **true**, the toolStripButton's ControlType and Role is changed to checkBox, and toolStripButton's controlType is button with **False**  Checked and  CheckOnClick properties, is it expected result?
![image](https://github.com/dotnet/winforms/assets/26474449/0a0753d6-30e1-4910-a9ad-eef0a63c037a)

**Comment by John-Qiao** [comment]
Verified this on .NET 8.0 Preview 5.0 TP build: 8.0.100-preview.5.23276.4, issue was fixed as same as above result that: There's no error in AccessibilityInsight test result, but in Inspect testing result, when toolStripButton's CheckOnClick or Checked property is true, the toolStripButton's ControlType and Role is changed to checkBox.

**Comment by dmitrii-drobotov** [comment]
@Olina-Zhang yes, this is expected.

**Comment by John-Qiao** [comment]
Verified this on .NET 8.0 TP build: 8.0.100-preview.5.23276.4, issue was fixed, test result is same as above test result.
